DVCodingStyle.md

```diff
@@ -594,30 +594,120 @@ cause race conditions if there are any parallel threads.
-    `raise_objection` or `drop_objection` within a sequence.
-5.  When starting a sequence that can raise and drop objections, if you want
-    the sequence to raise and drop objections, set the `starting_phase` member of
-    the sequence object the `<sequence_handle>.set_starting_phase(<uvm_phase>)`
-    before starting the sequence.
-6.  To avoid reimplementing objection functionality in multiple sequence classes,
-    create a parent sequence that implements this functionality for its derivatives.
-7.  When calling `raise_objection` or `drop_objection`, you must pass a string
+
+At a high level, a phase should not end when either there is more stimulus to
+send during that phase or the DUT has yet to respond to some stimulus that has
+been sent.
+
+To prevent a phase from ending in the former case, it is required to raise and
+lower objections inside the test class.
+
+To prevent a phase from ending in the latter case, it is required to raise and
+lower objections inside the `phase_ready_to_end()` method of classes derived
+from `uvm_component`.
+It is also recommended to only include objections in the monitor component.
+When using this approach, these guidelines should be followed:
+
+1.  The monitor's `phase_ready_to_end()` method should implement a watchdog timer
+    that raises an objection if any traffic is seen on the bus and lowers it once
+    no traffic is seen within the timeout period; if traffic is seen, the
+    watchdog resets its timer.
+    Refer to the example below:
+
+    ```systemverilog
+    function void phase_ready_to_end(uvm_phase phase);
+     string str;
+     uvm_object objectors[$];
+
+     // This example code assumes existence of two class members:
+     // 1) `bit ok_to_end`
+     // 2) `int timeout_delay`
+
+     // Only start the watchdog if the main phase is ending
+     if (phase.is(uvm_run_phase::get())) begin
+       // Start the watchdog - it will set the ok_to_end bit if no transactions
+       // have been seen on the bus.
+       fork
+         watchdog_ok_to_end();
+       join_none;
+       if (!ok_to_end) begin
+         // This objection will be raised if the watchdog detects traffic
+         // on the bus.
+         phase.raise_objection(this, $sformatf("%s objection raised", `gfn);
+         fork
+           begin
+             // Poll the ok_to_end bit until it goes high to signal that the
+             // previously raised objection can now be safely dropped.
+             wait_for_ok_to_end();
+             phase.drop_objection(this, $sformatf("%s objection dropped", `gfn);
+             // Once this point has been reached, it is a good idea to check if
+             // any other objects are still objecting.
+             phase.get_objection().get_objectors(objectors);
+             foreach(objectors[i]) begin
+               // Print information about any objects that
+               // still have raised objections.
+               ...
+             end
```

Comment:
I think you can leave it at the suggestion above.

Reply:
Yup, replaced all of this with a comment.